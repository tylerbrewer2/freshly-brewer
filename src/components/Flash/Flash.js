import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

// Style Helpers
import Colors from '../../styles/helpers/Colors';
import Fonts from '../../styles/helpers/Fonts';

const FlashCore = ({ children, className }) => (
  <section className="col-xs-12">
    <div className={className}>
      {children}
    </div>
  </section>
)

FlashCore.propTypes = {
  className: PropTypes.string.isRequired,
  children: PropTypes.string.isRequired,
  status: PropTypes.string.isRequired,
}

const Flash = styled(FlashCore)`
  font-family: ${Fonts.body};
  border: 1px solid ${props => props.status === 'success' ?  Colors.success : Colors.error};
  color: ${props => props.status === 'success' ? Colors.success : Colors.error };
  margin-top: 15px;
  text-align: left;
  border-radius: 3px;
  padding: 10px 15px;
  line-height: 25px;
`

export default Flash
