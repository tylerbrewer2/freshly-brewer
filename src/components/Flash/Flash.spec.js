import React from 'react';
import renderer from 'react-test-renderer';

import Flash from './Flash';

describe('component structure', () => {
  it('renders successful flash correctly', () => {
    const tree = renderer.create(
      <Flash status="success">
        Successful request.
      </Flash>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders unsuccessful flash correctly', () => {
    const tree = renderer.create(
      <Flash status="error">
        There as an error during the request.
      </Flash>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

})