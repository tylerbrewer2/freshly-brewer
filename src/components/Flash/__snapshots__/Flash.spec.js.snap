exports[`component structure renders successful flash correctly 1`] = `
<section
  className="col-xs-12">
  <div
    className="sc-bdVaJa itbRGo">
    Successful request.
  </div>
</section>
`;

exports[`component structure renders unsuccessful flash correctly 1`] = `
<section
  className="col-xs-12">
  <div
    className="sc-bdVaJa eBdQSA">
    There as an error during the request.
  </div>
</section>
`;
